packages/web-components/fast-foundation/src/form-associated/form-associated.ts

```diff
@@ -607,6 +608,12 @@ export function FormAssociated<T extends ConstructableFormAssociated>(BaseCtor:
                     this.proxy.value = this.value;
                 }
 
+                if (this.hasAttribute("form")) {
+                    this.proxy.setAttribute("form", this.getAttribute("form") as string);
+                } else {
```

Comment:
This won't catch the case where the proxy is already attached and *then* the form attribute is set on the element. I'd recommend the following:
1. Add the `form` attribute to this `FormAssociated` class (you can probably just copy what Button is doing). Note [you'll need to call the decorator as a function](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/form-associated/form-associated.ts#L675).
2. Add a `formIdChanged()` method to the `FormAssociated` and in it, set the id on the proxy if it exists (similar to https://github.com/microsoft/fast/blob/master/packages/web-components/fast-foundation/src/form-associated/form-associated.ts#L418). 

Doing so will ensure that the proxy's form attribute stays in sync with the host element's form attribute and should address the above edge case.

Reply:
I couldn't get it to work with the changes you've proposed. I've been spending a bit too much time outside of my usual space, so I'm going to close this PR due to the edge case you mentioned. You're welcome to pick it up from here if this PR is helpful!